src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -31,11 +32,54 @@ constexpr float DEFAULT_QUEUE_PRIORITY = 1.0f;
 } // namespace
 
 namespace inexor::vulkan_renderer::wrapper {
+namespace {
+
+struct DeviceInfo {
+    VkPhysicalDevice physical_device;
+    VkPhysicalDeviceType type;
+    VkDeviceSize total_device_local;
+    bool presentation_supported;
+    bool swapchain_supported;
+};
+
+unsigned device_type_rating(const DeviceInfo &info) {
+    switch (info.type) {
+    case VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU:
+        return 2;
+    case VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU:
+        return 1;
+    default:
+        return 0;
+    }
+}
 
-bool Device::is_extension_supported(const VkPhysicalDevice graphics_card, const std::string &extension_name) {
-    assert(graphics_card);
-    assert(!extension_name.empty());
+bool is_device_suitable(const DeviceInfo &info, const VkPhysicalDeviceFeatures &) {
```

Comment:
```suggestion
bool is_device_suitable(const DeviceInfo &info, const VkPhysicalDeviceFeatures &/*features*/) {
```

Reply:
Will leave this one since that part still needs to be implemented.

Reply:
I implemented it.